packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/events_storage_adapter.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/drift/drift_tables.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/src/pinpoint/common/serializers.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:built_value/serializer.dart';
+
+/// Interface with underlying device storage using the [Drift] package
+/// Present interface for saving/retrieving PinpointEvents
+class EventStorageAdapter {
+  factory EventStorageAdapter(CachedEventsPathProvider? pathProvider) {
+    final db = DriftDatabaseJsonStrings(pathProvider);
```

Comment:
If I read the code correctly, the db writes happens only within `saveEvent` call, should open the connection within the function call, and also close when finish saving, instead of keep the connection open all the time?

Reply:
I think db writes will also be done on deleteEvents

Yeah perhaps I am a bit unfamiliar with runtime performance here.  Is it expensive to keep it open vs open/close it on use?

Typically we will be saving events every 10 seconds (auto submission of events).

Reply:
I was think that each time when we need to write the db, to invoke `DriftDatabaseJsonStrings()` which open the connection, and after write, invoke `db.close()`.

But Drift actually doesn't recommend creating multiple instances of DB which duplicates the `QueryExecutor` which may lead to race condition. So single instance (connection) through out an App session seems reasonable. 

Probably need to test in mobile, when minimize App and reopen, will the connection still work? (same for me to test in Storage use cases.)

Reply:
I mis spoke above.  There are 2 cases when a db write will occur: 

1) When user calls `sendEvent` - instead of sending the event we will write to the db. 

2) Every 10 seconds, we auto send events.  This means we will delete events from the db which I assume is a write as well I assume.

Reply:
Yeah good point, good to capture that in an integration test. 

In manual tests on iOS when the App is minimized and put to foreground again, I am still able to read from the data